internal/storage/bucket_handle.go

```diff
@@ -23,3 +25,24 @@ type bucketHandle struct {
 	gcs.Bucket
 	bucket *storage.BucketHandle
 }
+
+func (b *bucketHandle) DeleteObject(ctx context.Context, req *gcs.DeleteObjectRequest) (err error) {
+	obj := b.bucket.Object(req.Name)
+
+	// Switching to the requested generation of the object.
+	if req.Generation != 0 {
+		obj = obj.Generation(req.Generation)
+	}
+	// Putting condition that the object's MetaGeneration should match the requested MetaGeneration for deletion to occur.
+	if req.MetaGenerationPrecondition != nil && *req.MetaGenerationPrecondition != 0 {
+		obj = obj.If(storage.Conditions{MetagenerationMatch: *req.MetaGenerationPrecondition})
+	}
+
+	// Deleting object through Go Storage Client.
+	err = obj.Delete(ctx)
+	if err != nil {
+		return
+	}
+
+	return
```

Comment:
if err is nil or not, you are returning. hence remove line 43-46

Reply:
resolved